cli/src/graphlog.rs

```diff
@@ -31,6 +33,28 @@ pub enum Edge<T> {
     Missing,
 }
 
+impl<T> Edge<T> {
+    pub fn map<U>(self, f: impl FnOnce(T) -> U) -> Edge<U> {
+        match self {
+            Edge::Direct(inner) => Edge::Direct(f(inner)),
+            Edge::Indirect(inner) => Edge::Indirect(f(inner)),
+            Edge::Missing => Edge::Missing,
+        }
+    }
+}
+
+/// Used to convert a jj_lib GraphEdge into a graphlog Edge.
+pub fn convert_edge_type<N>(e: GraphEdge<N>) -> Edge<N> {
+    // TODO: Is it possible for the ClI to use the GraphEdge type directly?
+    // These two types seem to correspond exactly, only differing in minor
+    // implementation details.
```

Comment:
Probably yes. I think the `graphlog::Edge` type exists just for historical reasons. If you're okay, can you add a refactoring patch first?

Reply:
If it's alright, I'd prefer to do a follow-up PR that just contains the `Edge -> GraphEdge` refactoring? If it's important to you for it to all be in this PR, that's fine. It's just a bit easier for me personally to open a new PR after this one.

Reply:
No problem. These functions can be easily removed later.